hathitrust/feed
# DEV-1120: Metadata fields not copied

## What/why?

Under some circumstances, during validation of submitted items, we lose image metadata that we rely on in later stages of validation.

In this case, we were losing the field `Artist` during image remediation, which is required in a later stage, causing validation to fail.

## What is the fix?

Put Artist back in the metadata. There are also some tidyings. I've tried to group the commits thematically (and will squash them upon approval).

## Testing instructions

Run these commands in order. Look in https://github.com/hathitrust/feed/pull/127/commits to see the commits mentioned.

```
# setup
git clone https://github.com/hathitrust/feed
cd feed
git checkout DEV-1120                                 # Not sure if needed, but dont hurt.
git checkout 26a7bdf21a318b58cf9884ba1a5d97e7e5ae4ec2 # Includes fixture and test, but not the code fix.
docker compose build
docker compose up -d

# part 1: should fail
cp t/fixtures/volumes/bitonal_tiff.zip volumes_to_test/
docker compose run --rm validate
docker compose run --rm test perl t/local_ingest.t

# part 2: should work
git checkout 4905bfcc29209ca44606dd5ed363fc86f0316872 # Includes the code fix.
docker compose run --rm validate
docker compose run --rm test perl t/local_ingest.t
```

To **"fail"** means:

Output from `compose run --rm validate` containing `ERROR: Missing field value	objid: bitonal_tiff	namespace: test	file: 00000001.tif`

Output from `docker compose run --rm test perl t/local_ingest.t` ending with `Looks like you failed 1 test of 19.`

To **"work"** means: 

No such error message in the output.

## Acknowledgements

* I acknowledge that I can't say for sure that this is the best way / place to add the fix.
* I acknowledge that the new test is a bit weak.
